heisenbridge/__main__.py

```diff
@@ -572,6 +606,10 @@ async def run(self, listen_address, listen_port, homeserver_url, owner, safe_mod
             try:
                 config = await self.az.intent.get_account_data("irc", room_id)
 
+                if config.get("is_hidden_room", False):
+                    self.hidden_room_id_internal = room_id
```

Comment:
If this was a real room type like in the original revision you could make it a validation rule that if the configuration option is not enabled the room is invalidated and it goes into cleanup queue, then you wouldn't need this double variable hack. I'm not blocking merging because of it but it would be slightly cleaner to keep it as a room type so it can be collected for cleanup when the feature is disabled.

Reply:
I initially had it as proper room type, so it'll definitely be nicer. I'll see if it can be easily revived.